core/templates/pages/exploration-editor-page/services/exploration-warnings.service.ts

```diff
@@ -488,6 +531,53 @@ export class ExplorationWarningsService {
   updateWarnings(): void {
     this._updateWarningsList();
   }
+
+  redirectionIsValid(
+      sourceStateName: string, destStateName: string, links: GraphLink[]
+  ): boolean {
+    let distance = this.getDistanceToDestState(
+      sourceStateName, destStateName, links);
+    // Raise validation error if the creator redirects the learner
+    // back by more than MAX_CARD_COUNT_FOR_VALID_REDIRECTION cards.
+    return distance <= AppConstants.MAX_CARD_COUNT_FOR_VALID_REDIRECTION;
+  }
+
+  getDistanceToDestState(
+      sourceStateName: string, destStateName: string, links: GraphLink[]
+  ): number {
+    let distance = -1;
```

Comment:
Should we to rename to distanceToDestState?

Reply:
Done.